content/A take on empowering ActiveRecord.md

```diff
@@ -0,0 +1,82 @@
+# A take on empowering ActiveRecord
+
+I think we are quite underpowered when it comes to doing queries on Rails, having this in mind I proposed my opinion on how we could improve this and made a PR. I also wanted to share it here to know what others think and maybe get some traction.
+
+## The reasoning behind this idea
+
+The problem I was trying to solve is one I usually face, and I think all of us do, where I have to use raw SQL strings for simple queries like:
+
+```ruby
+Post.where('created_at > ?', 1.month.ago)
+```
+
+I won't get into why using strings is not ideal but here's a quick example of how this could go wrong:
+
+```ruby
+Post.where('created_at > ?', 1.month.ago).joins(:comments)
+# run this query and you'll an ActiveRecord::StatementInvalid (PG::AmbiguousColumn: ERROR:  column reference "created_at" is ambiguous)
+```
+
+Aside from that I really love Ruby and I'd like to write as much Ruby and as less of any other language as possible.
+
+## Querying with Ruby blocks
+
+My idea of how querying can be enhanced is simple: use a block to build the query just as if you were using `#select` or `#map`.  Here's a quick example on how I imagine that syntax looking:
+
+```ruby
+Post.where { |post| post.created_at.gt(1.month.ago) }
+```
+
+Wouldn't you agree that's much better writing/parsing SQL strings? If you know what Arel is, you'll notice that I purposely used its syntax when comparing values. If you don't know what Arel is, a quick explanation is that it's what ActiveRecord uses to build queries under the hood.
+
+I used Arel syntax because I think it's great and I find myself using it a lot in complex queries. I also believe it has become very stable and that it provides great power with a good syntax. Although we could also just alias some comparison methods and have something that looks much more natural:
```

Comment:
```suggestion
I used Arel syntax because I think it's great and I find myself using it a lot in complex queries. I also believe it has become very stable and that it provides great capabilities with a good syntax. Although we could also just alias some comparison methods and have something that looks much more natural:
```